tests/add-route.spec.ts

```diff
@@ -2,7 +2,11 @@ import { test, expect } from '@playwright/test';
 
 test('testing add a route and save it', async ({ page }) => {
     await page.goto('/scheme.html?authority=Derby#16.84/52.906457/-1.504519');
-    await page.getByRole('region', { name: 'Map' }).waitFor();
+    // wait for the map to load and interventions panel to appear
+    await page.getByText('Edit attributes Click an intervention to fill out its attributes Edit geometry N').waitFor();
```

Comment:
Playwright doesn't know to wait for the map to load or for route snapper to finish loading. So we add some explicit waits in here for text elements that relate to those features.

I'm open to a potentially better way of waiting for these things (spent a non-trivial amount of time trying to work out how to wait for the console log message from router-snapper but failed), but as a first cut these seem to behave.

Reply:
I haven't tried it out yet, but I checked maplibre's own tests for inspiration: https://github.com/birkskyum/maplibre-gl-js/blob/c82f26ff6e3b1d4fb735db2c5b71d654d3319972/test/integration/browser/browser.test.ts#L51
They await a DOM element that maplibre inserts: `const canvas = await page.$('.maplibregl-canvas');`
We may be able to try something similar. Worth noting they just switched from Playwright to Puppeteer, but I think this particular technique didn't change: https://github.com/maplibre/maplibre-gl-js/pull/2494

In the meantime, I'm happy waiting for text like this as a proxy. We control it, so we could use some other way to mark that the route snapper is loaded. Maybe it slightly violates the principle of black-box testing, but do something blunt like `window.routeSnappedReady = true` and wait for that.

But I would say just trim down the text to not include the buttons? It reads a bit strangely in the test. How about just
```suggestion
    await page.getByText('Click an intervention to fill out its attributes').waitFor();
```

Reply:
Started https://osmus.slack.com/archives/C01G3D28DAB/p1684002466097189 to ask about maplibre+playwright

Reply:
A quick reply from Slack! We could do this:

https://github.com/birkskyum/maplibre-gl-js/blob/c82f26ff6e3b1d4fb735db2c5b71d654d3319972/test/integration/browser/browser.test.ts#LL32C1-L50C1

Reply:
... figuring out how to get ahold of `map`. Maybe something like `window.map = ...` in `Map.svelte` isn't the worst hack in the world

Reply:
> But I would say just trim down the text to not include the buttons? It reads a bit strangely in the test. How about just

This is code generated from codegen, I'll give it a try with your suggestion locally and see if I can include.

Reply:
Migrating the ideas about the map here to an issue to address in a new PR